src/Symfony/Component/Lock/Store/RetryTillSaveStore.php

```diff
@@ -54,11 +57,23 @@ public function save(Key $key)
         $this->decorated->save($key);
     }
 
+    /**
+     * {@inheritdoc}
+     */
+    public function putOffExpiration(Key $key, $ttl)
+    {
+        $this->decorated->putOffExpiration($key, $ttl);
+    }
+
     /**
      * {@inheritdoc}
      */
     public function waitAndSave(Key $key)
     {
+        if (!$this->decorated instanceof StoreInterface || (!$this->decorated instanceof BlockingStoreInterface || !$this->decorated->supportsWaitAndSave())) {
+            throw new NotSupportedException(sprintf('The store "%s" does not supports blocking locks.', \get_class($this->decorated)));
```

Comment:
Should be removed: The purpose of this store is to give the blocking behavior to the decorated service.